### PR TITLE
Make cowswap work in gnosis too

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/chain/gnosis/modules/cowswap/constants.py [hunk 1]
```diff
@@ -0,0 +1 @@
+CPT_COWSWAP = 'cowswap'
```
Comment (LefterisJP) [inline rotkehlchen/chain/gnosis/modules/cowswap/constants.py:1]: This should not be gnosis specific. Move this to the `evm/decoding/coswap/constants.py` file

File: rotkehlchen/tests/unit/decoders/test_cowswap.py [hunk 2]
```diff
@@ -436,3 +478,64 @@ def test_refund_eth_order(database, ethereum_inquirer, ethereum_accounts):
         ),
     ]
     assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('gnosis_accounts', [['0xc37b40ABdB939635068d3c5f13E7faF686F03B65']])
+def test_swap_gnosis_tokens(database, gnosis_inquirer, gnosis_accounts):
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/decoders/test_cowswap.py:485]: Could you maybe find some more tests for the other cases?

File: rotkehlchen/chain/evm/decoding/cowswap/decoder.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+            wrapped_native_asset: Asset,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.native_asset = native_asset.resolve_to_crypto_asset()
+        self.wrapped_native_asset = wrapped_native_asset.resolve_to_evm_token()
+        self.settlement_address = GPV2_SETTLEMENT_ADDRESS
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/cowswap/decoder.py:68]: yes love them

File: rotkehlchen/chain/evm/decoding/cowswap/decoder.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+            to_token_address = hex_or_bytes_to_address(tx_log.data[32:64])
+            raw_from_amount = hex_or_bytes_to_int(tx_log.data[64:96])
+            raw_to_amount = hex_or_bytes_to_int(tx_log.data[96:128])
+            raw_fee_amount = hex_or_bytes_to_int(tx_log.data[128:160])
+
+            if (
+                from_token_address == self.wrapped_native_asset.evm_address and
+                owner_address == self.native_asset_flow_address
+            ):
+                from_asset = self.native_asset  # native asset swaps are made by eth flow contract in cowswap  # noqa: E501
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/cowswap/decoder.py:135]: Yes 
image

File: rotkehlchen/chain/gnosis/modules/cowswap/constants.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/chain/gnosis/modules/cowswap/constants.py:1]: Is there two. I copy-pasted too much

File: rotkehlchen/chain/evm/decoding/cowswap/decoder.py [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+TRADE_SIGNATURE = b'\xa0zT:\xb8\xa0\x18\x19\x8e\x99\xca\x01\x84\xc9?\xe9\x05\ny@\n\nr4A\xf8M\xe1\xd9r\xcc\x17'  # noqa: E501
+PLACE_ETH_ORDER_SIGNATURE = b"\xcf_\x9d\xe2\x98A2&R\x03\xb5\xc35\xb2W\'p,\xa7rb\xffb.\x13k\xaasb\xbf\x1d\xa9"  # noqa: E501
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/cowswap/decoder.py:43]: Doesn't have a name, it looks like this 
image

File: rotkehlchen/chain/evm/decoding/cowswap/decoder.py [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+TRADE_SIGNATURE = b'\xa0zT:\xb8\xa0\x18\x19\x8e\x99\xca\x01\x84\xc9?\xe9\x05\ny@\n\nr4A\xf8M\xe1\xd9r\xcc\x17'  # noqa: E501
+PLACE_NATIVE_ASSET_ORDER_SIGNATURE = b"\xcf_\x9d\xe2\x98A2&R\x03\xb5\xc35\xb2W\'p,\xa7rb\xffb.\x13k\xaasb\xbf\x1d\xa9"  # noqa: E501
+REFUND_NATIVE_ASSET_ORDER_SIGNATURE = b'\x19Rq\x06\x8a(\x81\x91\xe4\xb2e\xc6A\xa5k\x982\x91\x9fi\xe9\xe7\xd6\xc2\xf3\x1b\xa4\x02x\xae\xb8Z'  # noqa: E501
+INVALIDATE_ETH_ORDER_SIGNATURE = b'\xb8\xba\xd1\x02\xac\x8b\xba\xcf\xef1\xff\x1c\x90n\xc6\xd9Q\xc20\xb4\xdc\xe7P\xbb\x03v\xb8\x12\xad5\x85*'  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/cowswap/decoder.py:45]: you changed the rest. Why not change this too?

**Review by LefterisJP — APPROVED** [review]
LGTM. Thank you

**Pull request merged**